packages/shell-api/bin/generate-metadata.ts

```diff
@@ -378,17 +379,17 @@ async function generateMetadata(
   const src = path.join(root, 'src');
   const opts = path.join(root, 'tsconfig.json');
 
-  const files = (await fs.readdir(src))
+  const files = (await promisify(fs.readdir)(src))
     .filter((name) => /\.ts$/.test(name) && !/\.spec/.test(name))
     .map((name) => path.join(src, name));
 
   const config: ts.CompilerOptions = JSON.parse(
-    await fs.readFile(opts, 'utf-8')
+    await promisify(fs.readFile)(opts, 'utf-8')
   );
 
   const metadata = await generateMetadata(files, config);
 
-  await fs.writeFile(
+  await promisify(fs.writeFile)(
```

Comment:
Just curious, we use `fs.promises` a lot in mongosh, and the only release lines without `fs.promises` but with `promisify` are 8.x & 9.x, which have been EOL for a while, what’s the situation where we can’t use `fs.promises`?

Reply:
Ohhh, yeah, you're totally right, we use it a lot and it doesn't fail. I refactored from failing `fs/promises` import that was added only in 14 directly to using `promisify`, totally skipped my mind that `fs.promises` was added before `fs/promises`. I'll change it to `fs.promises` as it's definitely a better option here